### DISCUSSION TITLE
Time-Weighted Averages Blog Questions & Feedback

### DISCUSSION THREAD

**Discussion by davidkohn88** [timescale/timescaledb-toolkit#185] [General]
This is a discussion for folks who have questions / comments / feedback on our recent blogpost on time-weighted averages: https://blog.timescale.com/blog/what-time-weighted-averages-are-and-why-you-should-care/

**Comment by abdelkaderralem** [comment]
Hi Everyone, 
I have read "What Time-Weighted Averages Are and Why You Should Care" post.
I m doing simulation about TCP protocol.
I have a two column data time and the value for the instant throughput of tcp.
could i use the function to calculate the time-weighted average for throughput, and that for every instance when the value change.
as a result i will get the time-weighted average of TCP throughput. 
Thank You.

**Comment by Timsgmlr** [comment]
I'm a little confused about the behavior of time weighted averages using LOCF. Let's say I have following data points: 
| Time | Value |
| ------------- | ------------- |
| 2020-01-01 00:00:00  | 1  |
| 2020-01-01 23:00:00  |1000  |

When calculating the weighted average for a 24 hour interval my calculation would be: `1 * 23/24 + 1000 * 1/24 = 42.625`. 
When doing the same thing with timescaledb using following query: 
```
SELECT time_bucket('1 days', time) AS timebucket,
       average(time_weight(
               'locf',
               time,
               value
           ))                      
FROM series_points_float
WHERE (time BETWEEN TIMESTAMP '2020-01-01 00:00:00+00:00' AND TIMESTAMP'2020-01-02 00:00:00+00:00')
GROUP BY timebucket
```

The result is just 1. Can someone please point out why the second data point is not taken into consideration?
